### PR TITLE
Task 4 (Vampire)

### PR DESCRIPTION
**Pull request by DmitriyVolik**

### CODE REVIEW

File: WarOOP.Tests/WarriorTests.cs [hunk 1]
```diff
@@ -45,4 +46,33 @@ public void Fight_WarriorAndDefender_Correct()
...
+    [Fact]
+    public void AttackTo_VampireToWarrior_Correct()
+    {
+        var warrior = new Warrior();
+        var vampire = new Vampire();
+
+        warrior.AttackTo(vampire);
+        vampire.AttackTo(warrior);
+        var result = vampire.CurrentHealth ==
+                     vampire.StartHealth - warrior.Attack + vampire.Attack * vampire.Vampirism / 100;
```
Comment (elkavitaly) [inline WarOOP.Tests/WarriorTests.cs]: Move this expression to separate variable to make the purpose of this expression more clear

File: WarOOP/Models/Vampire.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+namespace WarOOP.Models;
+
+public class Vampire : Warrior
+{
+    public readonly int Vampirism;
```
Comment (elkavitaly) [inline WarOOP/Models/Vampire.cs]: Don't make fields public. This field is used only for internal logic, so you don't need to expose it at all.

File: WarOOP/Models/Warrior.cs [hunk 3]
```diff
@@ -17,16 +17,19 @@ public Warrior()
...
     }
 
-    protected virtual void GetDamageFrom(Warrior enemy)
+    protected internal virtual int GetDamageFrom(Warrior enemy)
     {
         if (IsAlive)
         {
             CurrentHealth -= enemy.Attack;
+            
+            return enemy.CurrentHealth < 0 ? enemy.Attack + enemy.CurrentHealth : enemy.Attack;
```
Comment (elkavitaly) [inline WarOOP/Models/Warrior.cs:26]: Why do you return the result of this expression `enemy.Attack + enemy.CurrentHealth`?

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: WarOOP/Models/Warrior.cs (hunk 3 above)
Comment (DmitriyVolik) [reply WarOOP/Models/Warrior.cs:26]: It's necessary when enemy health < 0, in this case it returns only attack down to zero

File: WarOOP/Models/Vampire.cs (hunk 2 above)
Comment (DmitriyVolik) [reply WarOOP/Models/Vampire.cs]: I need it for tests, but i can add private setter

**Review by elkavitaly — APPROVED** [review]

**Pull request merged**